src/pe/header.rs

```diff
@@ -396,73 +407,300 @@ impl Default for DosStub {
+    #[doc(alias("Machine"))]
     pub machine: u16,
+    /// The number of sections. This indicates the size of the section table, which immediately follows the headers.
+    /// Note that the Windows loader limits the number of sections to 96.
+    /// [Source](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_file_header).
+    #[doc(alias("NumberOfSections"))]
     pub number_of_sections: u16,
+    /// The low 32 bits of the time stamp of the image. This represents the date and time the image was created by the linker.
+    /// The value is represented in the number of seconds elapsed since midnight (00:00:00), January 1, 1970, Universal
+    /// Coordinated Time, according to the system clock.
+    #[doc(alias("TimeDateStamp"))]
     pub time_date_stamp: u32,
+    /// The offset of the symbol table, in bytes, or zero if no COFF symbol table exists.
+    ///
+    /// Typically, this field is set to 0 because COFF debugging information is deprecated.
+    /// [Source](https://0xrick.github.io/win-internals/pe4/#file-header-image_file_header).
+    // TODO: further explain the COFF symbol table. This seems to be a nuanced topic.
+    #[doc(alias("PointerToSymbolTable"))]
     pub pointer_to_symbol_table: u32,
+    /// The number of symbols in the symbol table.
+    ///
+    /// Typically, this field is set to 0 because COFF debugging information is deprecated.
+    /// [Source](https://0xrick.github.io/win-internals/pe4/#file-header-image_file_header).
+    // Q (JohnScience): Why is the name `number_of_symbol_table` and not `number_of_symbols`?
+    #[doc(alias("NumberOfSymbols"))]
     pub number_of_symbol_table: u32,
+    /// The size of the optional header, in bytes. This value should be zero for object files.
+    ///
+    /// The [`goblin::pe::optional_header::OptionalHeader`](crate::pe::optional_header::OptionalHeader) is meant to
+    /// represent either the 32-bit or the 64-bit optional header. The size of the optional header is used to determine
+    /// which one it is.
+    #[doc(alias("SizeOfOptionalHeader"))]
     pub size_of_optional_header: u16,
+    /// The [characteristics](https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#characteristics) of the image.
+    ///
+    /// The constants for the characteristics are available in the [`goblin::pe::characteristic`](crate::pe::characteristic) module.
+    #[doc(alias("Characteristics"))]
     pub characteristics: u16,
 }
 
 pub const SIZEOF_COFF_HEADER: usize = 20;
 /// PE\0\0, little endian
 pub const PE_MAGIC: u32 = 0x0000_4550;
 pub const SIZEOF_PE_MAGIC: usize = 4;
-/// The contents of this field are assumed to be applicable to any machine type
+
+// Q (JohnScience): doesn't it make sense to move all these constants to a dedicated module
+// and then re-export them from here? This way, the module will be more organized.
+//
+// Also, don't we want to declare them in a macro to remove the boilerplate and make the implementation
```

Comment:
file a issue if you feel theres room for improvement :)

Reply:
@m4b I will!